File: src/frontend/types/customer.ts
export interface Customer {
  id: string;
  name: string;
  email?: string;
  phone: string;
  address?: {
    street: string;
    city: string;
    state: string;
    zip: string;
  };
  propertySize?: number; // in square feet
  tags: string[];
  status: "active" | "inactive" | "lead";
  createdAt: Date;
  updatedAt: Date;
  lastContactedAt?: Date;
  totalJobsCompleted: number;
  totalRevenue: number;
  notes?: string;
  preferredContactMethod?: "sms" | "email" | "phone";
}

export interface CustomerProfile extends Customer {
  serviceHistory: ServiceHistory[];
  upcomingAppointments: Appointment[];
  estimates: Estimate[];
}

export interface ServiceHistory {
  id: string;
  customerId: string;
  serviceType: string;
  date: Date;
  cost: number;
  notes?: string;
  status: "completed" | "cancelled";
}

export interface Appointment {
  id: string;
  customerId: string;
  customerName: string;
  serviceType: string;
  scheduledDate: Date;
  duration: number; // in minutes
  status: "scheduled" | "in_progress" | "completed" | "cancelled";
  assignedTo?: string;
  notes?: string;
}

File: src/backend/schemas/customer.py
from uuid import UUID
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, EmailStr


class CustomerBase(BaseModel):
    name: str
    email: Optional[str] = None
    phone: str
    address_street: Optional[str] = None
    address_city: Optional[str] = None
    address_state: Optional[str] = None
    address_zip: Optional[str] = None
    property_size: Optional[int] = None
    tags: list[str] = []
    status: str = "active"
    preferred_contact_method: str = "sms"
    notes: Optional[str] = None


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    address_street: Optional[str] = None
    address_city: Optional[str] = None
    address_state: Optional[str] = None
    address_zip: Optional[str] = None
    property_size: Optional[int] = None
    tags: Optional[list[str]] = None
    status: Optional[str] = None
    preferred_contact_method: Optional[str] = None
    notes: Optional[str] = None


class CustomerRead(CustomerBase):
    id: UUID
    total_jobs_completed: int
    total_revenue: float
    created_at: datetime
    updated_at: datetime
    last_contacted_at: Optional[datetime] = None

    model_config = {"from_attributes": True}
